security/pkg/nodeagent/sds/monitoring.go

```diff
@@ -15,150 +15,40 @@
-		Namespace: "citadel_agent",
-		Subsystem: "sds_service",
-		Name:      "total_push_errors",
-		Help:      "The total number of failed SDS pushes.",
-	})
+	totalPushErrorCounts = monitoring.NewSum(
+		"total_push_errors",
+		"The total number of failed SDS pushes.",
+	)
 
 	// totalActiveConnCounts records total number of active SDS connections.
-	totalActiveConnCounts = prometheus.NewGauge(prometheus.GaugeOpts{
-		Namespace: "citadel_agent",
-		Subsystem: "sds_service",
-		Name:      "total_active_connections",
-		Help:      "The total number of active SDS connections.",
-	})
+	totalActiveConnCounts = monitoring.NewSum(
+		"total_active_connections",
+		"The total number of active SDS connections.",
+	)
 
 	// totalStaleConnCounts records total number of stale SDS connections.
-	totalStaleConnCounts = prometheus.NewGauge(prometheus.GaugeOpts{
-		Namespace: "citadel_agent",
-		Subsystem: "sds_service",
-		Name:      "total_stale_connections",
-		Help:      "The total number of stale SDS connections.",
-	})
-
-	// pendingPushPerConnCounts records the number of SDS requests in an active connection that are
-	// not responded yet. The label of a connection is represented as <resource name>-<connection ID>,
-	// and the value should be 0 or 1.
-	pendingPushPerConnCounts = prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Namespace: "citadel_agent",
-		Subsystem: "sds_service",
-		Name:      "pending_push_per_connection",
-		Help:      "The number of active SDS connections which are waiting for SDS push.",
-	}, []string{"resourcePerConn"})
-
-	// staleConnCounts records all the stale connections which will be closed. The label of a
-	// stale connection is represented as <connection ID>, and the value should be 1.
-	staleConnCounts = prometheus.NewCounterVec(prometheus.CounterOpts{
-		Namespace: "citadel_agent",
-		Subsystem: "sds_service",
-		Name:      "stale_conn_count",
-		Help:      "The number of stale SDS connections.",
-	}, []string{"resourcePerConn"})
-
-	// pushPerConnCounts records the number of SDS pushes in an active connection. The label of a
-	// connection is represented as <resource name>-<connection ID>, and the value should be at
-	// least 1.
-	pushPerConnCounts = prometheus.NewCounterVec(prometheus.CounterOpts{
-		Namespace: "citadel_agent",
-		Subsystem: "sds_service",
-		Name:      "pushes_per_connection",
-		Help:      "The number of secret pushes to an active SDS connection.",
-	}, []string{"resourcePerConn"})
-
-	// pushErrorsPerConnCounts records the number of SDS push failures in an active connection.
-	// The label of a connection is represented as <resource name>-<connection ID>, and the value
-	// should be at least 1.
-	pushErrorsPerConnCounts = prometheus.NewCounterVec(prometheus.CounterOpts{
-		Namespace: "citadel_agent",
-		Subsystem: "sds_service",
-		Name:      "push_errors_per_connection",
-		Help:      "The number of failed secret pushes to an active SDS connection.",
-	}, []string{"resourcePerConn"})
-
```

Comment:
Could you please add a TODO to dump these metrics in debug endpoint?

Reply:
Sure. Will do.